File: ios/Feats/Feats/Models/Comment.swift
import Foundation

struct Comment: Codable, Identifiable, Equatable {
    let id: String
    let postId: String
    let userId: String
    let parentId: String?
    let content: String
    let createdAt: Date
    let updatedAt: Date
    let user: User?
    let replies: [Comment]?

    enum CodingKeys: String, CodingKey {
        case id, content, user, replies
        case postId = "post_id"
        case userId = "user_id"
        case parentId = "parent_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    static func == (lhs: Comment, rhs: Comment) -> Bool {
        lhs.id == rhs.id
    }
}

struct CreateCommentRequest: Codable {
    let content: String
    let parentId: String?

    enum CodingKeys: String, CodingKey {
        case content
        case parentId = "parent_id"
    }
}

struct UpdateCommentRequest: Codable {
    let content: String
}

File: backend/internal/models/comment.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Comment struct {
	ID        string         `gorm:"type:text;primaryKey" json:"id"`
	PostID    string         `gorm:"type:text;not null;index" json:"post_id"`
	UserID    string         `gorm:"type:text;not null" json:"user_id"`
	ParentID  *string        `gorm:"type:text;index" json:"parent_id,omitempty"`
	Content   string         `gorm:"type:text;not null" json:"content"`
	CreatedAt time.Time      `gorm:"type:datetime;not null" json:"created_at"`
	UpdatedAt time.Time      `gorm:"type:datetime;not null" json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"type:datetime;index" json:"-"`

	// Relationships
	Post    Post      `gorm:"foreignKey:PostID" json:"-"`
	User    User      `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Parent  *Comment  `gorm:"foreignKey:ParentID" json:"-"`
	Replies []Comment `gorm:"foreignKey:ParentID" json:"replies,omitempty"`
}

func (Comment) TableName() string {
	return "comments"
}

func (c *Comment) IsReply() bool {
	return c.ParentID != nil
}
